apps/system/js/pin_page_system_dialog.js

```diff
@@ -82,6 +104,13 @@
     this.header = document.querySelector(prefix + ' gaia-header');
     var pinSelector = prefix + ' button[data-action="pin"]';
     this.pinButton = document.querySelector(pinSelector);
+    this.arrow = document.getElementById('pin-arrow');
+    this.pinSiteContainer = document.getElementById('pin-site-container');
+    this.siteBadge = document.querySelector(prefix + ' gaia-app-icon');
+    this.siteName = document.querySelector(prefix + ' #site-name');
+    this.origin = document.querySelector(prefix + ' .origin');
+    pinSelector = prefix + ' button[data-action="pin-site"]';
+    this.pinSiteButton = document.querySelector(pinSelector);
```

Comment:
I understand you just copied the logic here, but what about something like that:
```js
this.siteBadge = this.element('gaia-app-icon');
```
Instead of `document.querySelector(prefix + ' #bla-bla');`.